### ISSUE TITLE
Techrep objects throw an error in edata_replace

### ISSUE THREAD

**Issue by rarichardson92** [pmartR/pmartR#102]
Peptide data with technical replicates seem to break edata_replace? Error can be reproduced with code below:

------------------------
```
library(pmartRdata)
data(techrep_pep_object)
techrep_pep_object2 <- edata_replace(omicsData = techrep_pep_object, x=0, y=NA)

### Odd since there seems to be no problem here:
techrep_pep_object$f_data$RunID %in% colnames(techrep_pep_object$e_data)

### Doesn't work for a newly constructed techrep object either:
techrep <- as.pepData(e_data = techrep_edata, f_data = techrep_fdata, edata_cname = "Mass_Tag_ID", fdata_cname = "RunID", techrep_cname = "TECH_REP", data_scale = "log2")

techrep2 <- edata_replace(omicsData = techrep, x=0, y=NA)
```

**Comment by clabornd** [comment]
check.names strikes again:  when I made this object I didn't set `attributes(omicsData)$check.names=FALSE` , which causes sadness within `edata_replace`.

Specifically, in edata_replace.R, when it creates the object `edata_new`, the columns get all screwed up which causes `as.pepData` to fail.

Also I neglected to hunt down all the `as.omicsData()` calls and make sure they are resetting the techrep_cname attribute.  At the very least, the object resulting from edata_replace() on an object with technical replicates removes `attributes(techrep_pep_object)$cnames$techrep_cname`

`library(pmartRdata)`
`data(techrep_pep_object)`
`attributes(techrep_pep_object)$check.names = FALSE`

`# runs without error....`
`techrep_pep_object2 <- edata_replace(omicsData = techrep_pep_object, x=0, y=NA)`
`#....but no techreps`
`attributes(techrep_pep_object2)$cnames$techrep_cname`
